applications/minotari_merge_mining_proxy/src/proxy/inner.rs

```diff
@@ -112,39 +112,53 @@ impl InnerService {
             ));
         }
 
-        let mut base_node_client = self.base_node_client.clone();
-        trace!(target: LOG_TARGET, "Successful connection to base node GRPC");
+        let monero_height = json["height"].as_u64().unwrap_or_default();
+        let monero_hash: Vec<u8> = Hex::from_hex(json["hash"].as_str().unwrap_or_default()).unwrap_or_default();
+        let mut base_node_height = 0;
+        let mut base_node_hash = vec![];
+        let mut p2pool_height = 0;
+        let mut p2pool_hash = vec![];
+
+        if let Some(mut p2pool_client) = self.p2pool_client.clone() {
+            let p2pool_resp = p2pool_client.get_tip_info(GetTipInfoRequest {}).await?;
+            let res = p2pool_resp.into_inner();
+
+            base_node_height = res.node_height;
+            base_node_hash = res.node_tip_hash.clone();
+            p2pool_height = res.p2pool_rx_height;
+            p2pool_hash = res.p2pool_rx_tip_hash.clone();
+        } else {
+            let mut base_node_client = self.base_node_client.clone();
+            trace!(target: LOG_TARGET, "Successful connection to base node GRPC");
 
-        let result =
-            base_node_client
-                .get_tip_info(tari_rpc::Empty {})
-                .await
-                .map_err(|err| MmProxyError::GrpcRequestError {
+            let result = base_node_client.get_tip_info(tari_rpc::Empty {}).await.map_err(|err| {
+                MmProxyError::GrpcRequestError {
                     status: err,
                     details: "get_tip_info failed".to_string(),
-                })?;
-        let height = result
-            .get_ref()
-            .metadata
-            .as_ref()
-            .map(|meta| meta.best_block_height)
-            .ok_or(MmProxyError::GrpcResponseMissingField("base node metadata"))?;
-        if result.get_ref().initial_sync_achieved != self.initial_sync_achieved.load(Ordering::SeqCst) {
-            self.initial_sync_achieved
-                .store(result.get_ref().initial_sync_achieved, Ordering::SeqCst);
-            debug!(
-                target: LOG_TARGET,
-                "Minotari base node initial sync status change to {}",
-                result.get_ref().initial_sync_achieved
-            );
```

Comment:
I think we should keep this

Reply:
Yes, the merge mining proxy now reports initial sync being achieved when it has not been:
```
2025-02-20 16:01:46.863171000 [minotari_mm_proxy::proxy::inner] DEBUG Initial base node sync achieved. Ready to mine at height #2277
```
vs.
```
16:01 v1.11.5-pre.0 esmeralda State: Syncing blocks: (dbbcb376e13725c5) 2766/7070 (39%)
```

Reply:
this piece of code merely sets the initial sync achieved atomicbool and it is set in a different place anyway

Reply:
I think its fine to remove this. 
We are mining invalid tari blocks yes, but we are not mining invalid monero blocks. We should not block the mining of monero while the tari block syncs.

Reply:
If we do not care about this, the config option to mine only after sync is achieved should be removed. As it currently stands, the logic does not work anymore.